### PR TITLE
[Reviewer: Matt] Set Cassandra memory usage in the clearwater-infrastructure script

### PR DESCRIPTION
**Pull request by rkd-msw**
Fixes https://github.com/Metaswitch/clearwater-infrastructure/issues/315

Rather than changing Cassandra's memory usage config once (in clearwater-aio-install.sh) which can then be overwritten, this generates that config along with the rest of the file in clearwater-infrastructure.

We can now also use `reduce_cassandra_mem_usage=Y` in other low-memory situations like t2.micro EC2 nodes or containers.

I've tested by:

* installing this package on an AIO node
* checking the config was right
* toggling the `reduce_cassandra_mem_usage` setting, re-running the infrastructure script, and checking it updated the file and restarted Cassandra
* not toggling the `reduce_cassandra_mem_usage` setting, re-running the infrastructure script, and checking didn't restart Cassandra

### CODE REVIEW

File: clearwater-cassandra/usr/share/clearwater/cassandra/cassandra-env.sh.template [hunk 1]
```diff
@@ -137,6 +137,7 @@ esac
 
 #MAX_HEAP_SIZE="4G"
 #HEAP_NEWSIZE="800M"
+TEMPLATE_HEAPSIZE_SECTION
```
Comment (mirw) [inline clearwater-cassandra/usr/share/clearwater/cassandra/cassandra-env.sh.template]: Should we make it clear that these are sections that will be replaced in the final file?  As it is, they look like incomplete variable definitions.  For example, changing them to `<<<TEMPLATE_HEAPSIZE_SECTION>>>`?

**Comment by mirw** [comment]
Looks good - 1 comment inline, and then a couple of other thoughts below:

*   How do we know that clearwater-infrastructure will always be run when a new version of cassandra is applied?  (Or is this still a gap, albeit one we might not care about?)

*   What happened to us moving away from modifying package-owned files (or at least telling dpkg that we were doing so)?

**Comment by rkd-msw** [comment]
We've already told dpkg about it - https://github.com/Metaswitch/clearwater-cassandra/blob/master/debian/clearwater-cassandra.hide#L12 makes /etc/cassandra/cassandra-env.sh not be owned by Cassandra - it only exists because it's generated by our infrastructure script.

That means clearwater-infrastructure doesn't have to be rerun on a cassandra upgrade, only on a clearwater-cassandra upgrade, which is done by clearwater-cassandra.postinst.

**Comment by mirw** [comment]
Great, thanks!

**Pull request merged**